google-cloud-storage/src/test/java/com/google/cloud/storage/ITGrpcReadTimeoutTest.java

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.storage;
+
+import static com.google.cloud.storage.TestUtils.apiException;
+import static com.google.cloud.storage.TestUtils.getChecksummedData;
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.Assert.assertThrows;
+
+import com.google.api.gax.retrying.RetrySettings;
+import com.google.api.gax.rpc.ServerStream;
+import com.google.api.gax.rpc.WatchdogTimeoutException;
+import com.google.common.base.Stopwatch;
+import com.google.protobuf.ByteString;
+import com.google.storage.v2.ContentRange;
+import com.google.storage.v2.Object;
+import com.google.storage.v2.ReadObjectRequest;
+import com.google.storage.v2.ReadObjectResponse;
+import com.google.storage.v2.StorageClient;
+import com.google.storage.v2.StorageGrpc;
+import com.google.storage.v2.StorageSettings;
+import io.grpc.Status.Code;
+import io.grpc.stub.StreamObserver;
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.concurrent.TimeUnit;
+import org.junit.Test;
+import org.threeten.bp.Duration;
+
+/**
+ * ReadObject leverages gRPC ServerStream to read a stream of ReadObjectResponse messages spanning
+ * the size of the object. If a large object is downloaded and it would take longer than the
+ * totalTimeout from {@link StorageOptions#getRetrySettings()}{@link RetrySettings#getTotalTimeout()
+ * #getTotalTimeout()}, gax will interrupt the stream with a DEADLINE_EXCEEDED error.
+ *
+ * <p>Instead of relying on total stream timeout, we rely on idleTimeout for the stream via {@link
+ * com.google.api.gax.rpc.ServerStreamingCallSettings.Builder#setIdleTimeout(Duration)}.
+ *
+ * <p>These tests force specific timeout scenarios to happen against an in-process grpc server to
+ * ensure our configuration of the StorageClient properly translates to the behavior we want.
+ *
+ * <p><i>NOTE:</i>Unfortunately, these tests are slow as they are waiting on wall clock time in
+ * several circumstances.
+ */
+public final class ITGrpcReadTimeoutTest {
```

Comment:
Nit this looks more like a conformance test or unit test and not an integration test similar to our existing IT which hit the live service.

Reply:
I can drop the IT prefix. I think it's probably dangling there from the first attempt trying to use testbench.

Reply:
Remove the dangling `IT`